packages/web/components/Dashboard/LatestContentTabs/Watch.tsx

```diff
@@ -1,8 +1,107 @@
-import { Box, Text } from '@metafam/ds';
-import React from 'react';
-
-export const Watch: React.FC = () => (
-  <Box>
-    <Text>Feed of MetaMedia YouTube content</Text>
-  </Box>
-);
+import { AspectRatio, Box, LoadingState } from '@metafam/ds';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+
+import { useOnScreen } from '../../../lib/hooks/useOnScreen';
+import { LIMIT, URL } from '../../../utils/LatestContentHelpers';
+
+type Thumbnail = {
+  url: string;
+  width: number;
+  height: number;
+};
+interface Video {
+  etag: string;
+  id: string;
+  kind: string;
+  snippet: {
+    channelId: string;
+    channelTitle: string;
+    description: string;
+    playlistId: string;
+    position: number;
+    publishedAt: string;
+    resourceId: {
+      kind: string;
+      videoId: string;
+    };
+    title: string;
+    videoOwnerChannelId: string;
+    videoOwnerChannelTitle: string;
+    thumbnails: {
+      default?: Thumbnail;
+      high?: Thumbnail;
+      medium?: Thumbnail;
+      standard?: Thumbnail;
+      maxres?: Thumbnail;
+    };
+  };
+}
+
+export const Watch: React.FC = () => {
+  const [videos, setVideos] = useState<Array<Video> | []>([]);
+  const [nextPageToken, setNextPageToken] = useState<string | null>(null);
+  const moreRef = useRef<HTMLDivElement | null>(null);
+  const [count, setCount] = useState<number>(0);
+  const [page, setPage] = useState<number>(1);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [isFetchingMore, setIsFetchingMore] = useState<boolean>(false);
+
+  const onScreen = useOnScreen(moreRef);
+
+  const maxPages = useMemo(() => {
+    const pages = Math.ceil(count / LIMIT) || 0;
+    return pages;
+  }, [count]);
+
+  useEffect(() => {
+    const load = async () => {
+      const res = await fetch(URL).then((r) => r.json());
+
+      setNextPageToken(res.nextPageToken);
+      setVideos(res.items);
+      setCount(res.pageInfo.totalResults);
+      setLoading(false);
+    };
+    load();
+  }, []);
+
+  useEffect(() => {
+    if (!loading && onScreen && page <= maxPages && nextPageToken) {
+      setIsFetchingMore(true);
+      const fetchMoreURL = `${URL}&pageToken=${nextPageToken}`;
+
+      fetch(fetchMoreURL)
```

Comment:
can you refactor that to use the `useSwr` package?

Reply:
ok i will have refactor in `useSwrInfinite` but i am not sure if i should do it now or after we build component that would display read,watch and listen combined.